### ISSUE TITLE
How to implement federated transfer learning?

### ISSUE THREAD

**Issue by chinatzsj** [tensorflow/federated#863]
I'm trying to implement federated transfer learning described at https://arxiv.org/pdf/1812.03337.pdf

It involves that two parties hold two neural networks respectively, compute the inner product of the output vector of the two networks and define loss function based on this.

But the tensorflow federated seems not to support such kind of construction, since the federated computing between the values held by two parties is limited to federated sum, federated map, federated reduce, etc.

So I am wondering whether the desired architecture could be achieved?

**Comment by jkr26** [comment]
Hi @chinatzsj 

This seems like a usage question of TFF, as opposed to a question about the code proper. Would you mind reposting to StackOverflow with the tag tensorflow-federated for discoverability? TFF has options for situations like this, including parameterizing [`federated_aggregate`](https://www.tensorflow.org/federated/api_docs/python/tff/federated_aggregate) by an appropriate dot product, or using [`federated_collect`](https://www.tensorflow.org/federated/api_docs/python/tff/federated_collect) if necessary, but I hesitate to give a more complete answer here, and will await the post on SO.

Thanks!
